domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutEntityValidatorImpl.java

```diff
@@ -34,6 +39,34 @@ public ValidationResult validate(GenericLayoutEntity entity) {
                         (entity.getCaseType() != null ? entity.getCaseType().getReference() : "")
                     ), entity)
             );
+        } else {
+            if (entity.getCaseFieldElementPath() != null) {
+                if (entity.getCaseField().isComplexFieldType() || entity.getCaseField().isCollectionFieldType()) {
+                    List<CaseFieldEntity> caseFields = entity.getCaseType().getCaseFields();
+
+                    List<String> allPaths = CaseFieldEntityUtil.buildDottedComplexFieldPossibilities(caseFields);
```

Comment:
can we avoid hardcoding CaseFieldEntityUtil and use dependency injection instead?